tests/genai/judges/optimizers/test_dspy_utils.py

```diff
@@ -0,0 +1,132 @@
+"""Tests for DSPy utility functions."""
+
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+
+from mlflow.exceptions import MlflowException
+from mlflow.genai.judges.optimizers.dspy_utils import (
+    agreement_metric,
+    create_dspy_signature,
+    trace_to_dspy_example,
+)
+
+
+def test_sanitize_judge_name(sample_trace_with_assessment):
+    """Test judge name sanitization in trace_to_dspy_example."""
+    # The sanitization is now done inside trace_to_dspy_example
+    # Test that it correctly handles different judge name formats
+    from mlflow.genai.judges.optimizers.dspy_utils import trace_to_dspy_example
+
+    # Mock dspy module
+    mock_dspy = MagicMock()
+    mock_example = MagicMock()
+    mock_example.with_inputs.return_value = mock_example
+    mock_dspy.Example.return_value = mock_example
+
+    with patch.dict("sys.modules", {"dspy": mock_dspy}):
+        # Test with different case variations - they should all find the assessment
+        # The assessment name in the fixture is "mock_judge" in lowercase
+        assert trace_to_dspy_example(sample_trace_with_assessment, "  mock_judge  ") is not None
+        assert trace_to_dspy_example(sample_trace_with_assessment, "Mock_Judge") is not None
+        assert trace_to_dspy_example(sample_trace_with_assessment, "MOCK_JUDGE") is not None
+
+
+def test_trace_to_dspy_example_success(sample_trace_with_assessment):
+    """Test successful conversion of trace to DSPy example."""
+    pytest.importorskip("dspy", reason="DSPy not installed")
+
+    # Use the fixture directly
+    trace = sample_trace_with_assessment
+
+    # Use real DSPy since we've skipped if it's not available
+    result = trace_to_dspy_example(trace, "mock_judge")
+
+    assert result is not None
+    # Verify the result has the expected DSPy structure
+    assert hasattr(result, "inputs")
+    assert hasattr(result, "outputs")
+    assert hasattr(result, "result")
+    assert hasattr(result, "rationale")
```

Comment:
Can we assert `isinstance(result, dspy.Example)` and make more specific assertions about the contents of inputs / outputs / result / rationale?